src/stream/double_ended/mod.rs

```diff
@@ -0,0 +1,201 @@
+mod next_back;
+mod nth_back;
+mod rfind;
+mod rfold;
+mod try_rfold;
+
+use next_back::NextBackFuture;
+use nth_back::NthBackFuture;
+use rfind::RFindFuture;
+use rfold::RFoldFuture;
+use try_rfold::TryRFoldFuture;
+
+extension_trait! {
+    use crate::stream::Stream;
+
+    use std::pin::Pin;
+    use std::task::{Context, Poll};
+
+
+    #[doc = r#"
+    Something fancy
```

Comment:
```suggestion
            Something fancy
```
Although it is a detailed place